src/contracts/aludel.ts

```diff
@@ -52,6 +56,139 @@ export async function getNetworkStats(signer: any) {
     lastUpdate: lastUpdate.toNumber(),
   };
 }
+function getFutureTotalStakeUnits(currentTotalStakeUnits: any, currentTotalStake: any, timestamp: number){
+  return currentTotalStakeUnits.add(currentTotalStake.mul(timestamp))
+}
+export async function getChartData(signer: any, amount: any, stakingStart:any) {
+  console.log("Amount", amount.toString(), stakingStart)
+  console.log("Amouunt", FixedNumber.from(amount), FixedNumber.from(amount).toString())
+
+  const aludel = new ethers.Contract(args.aludel, aludelAbi, signer);
+  const bonusMistToken = new ethers.Contract(args.mist, IERC20.abi, signer);
+  const wethRewardToken = new ethers.Contract(args.weth, IERC20.abi, signer);
+  const rewardScalingPeriod = 60 * 24 * 60 * 60;
+  const floor = 1;
+  const ceiling = 10;
+  const now = Math.floor(Date.now() / 1000); // Replace with staking period start
+  // Total rewards pool
+  let [mistRewardsBalance, wethRewardsBalance, aludelData] = await Promise.all([
+    bonusMistToken.balanceOf(args.rewardPool), //returns BN
+    wethRewardToken.balanceOf(args.rewardPool), //returns BN
+    aludel.getAludelData(), // 5th or 7th arg
+  ])
+  console.log("Aludel data", aludelData)
+
+  let chartData = [];
+    // Get rewards and stake units at future time (days elapsed)
+  let sharesOutstanding = aludelData[4]
+  let currentTotalStakeUnits = aludelData[6]
+  let currentTotalStake = aludelData[5]
+  let scheduleShares = aludelData[8][0][2] // assumes 1 schedule
+  let lastUpdate = now;
+
+  for (let day = 0; day <= 60; day += 3) {
+    const daysAdditionalInSecs =  day * 60 * 60 * 24;
+    const stakeDuration = now + daysAdditionalInSecs - stakingStart; 
+    const totalUnlockedWeiRewards = getFutureUnlockedRewards(daysAdditionalInSecs, wethRewardsBalance, sharesOutstanding, scheduleShares); //returns BN
+    currentTotalStakeUnits = getFutureTotalStakeUnits(currentTotalStakeUnits, currentTotalStake, now + daysAdditionalInSecs - lastUpdate)
+    lastUpdate = now + daysAdditionalInSecs
```

Comment:
Similar to `_aludel.lastUpdate`. This timestamp is needed for `getFutureTotalStakeUnits`.